kano_video/ui/video.py

```diff
@@ -56,11 +57,14 @@ def __init__(self, e, playlist_name=None, permanent=False):
         self.add(button_grid)
 
         img = Gtk.Image()
-
+        
         if e['thumbnail']:
-            thumbnail = '{}/video_{}.jpg'.format(tmp_dir, time())
-            download_url(e['thumbnail'], thumbnail)
-            img.set_from_file(thumbnail)
+            if os.path.exists(e['thumbnail']):# use local thumbnail if one exists
+                img.set_from_file(e['thumbnail'])
+            else: #download the file if local does not exist
+                thumbnail = '{}/video_{}.jpg'.format(tmp_dir, time())
+                download_url(e['thumbnail'], thumbnail)
+                img.set_from_file(thumbnail)
```

Comment:
Ah, okay. Can we make this process a little more resilient with something along the lines of:

```python
thumbnail = os.path.join(image_dir, 'icons/no_thumbnail.png')

if e['thumbnail']:
    if os.path.exists(e['thumbnail']:
        thumbnail = e['thumbnail']
    else:
        download_path = os.path.join(tmp_dir, 'video_{}.jpg'.format(time()))
        success, err = download_url(e['thumbnail'], download_path)
        if success:
            thumbnail = download_path

img.set_from_file(thumbnail)
```

Also, at some point I think that we need to be caching both the downloaded images and, potentially, the YouTube API queries.

@cwilson-github What do you think?